### PR TITLE
Cache staker ui packages metadata

### PR DESCRIPTION
**Pull request by pablomendezroyo**
Cache the stakers ui packages metadata to speed up the initial loading. Update the cache on every request in background

### CODE REVIEW

File: packages/dappmanager/src/db/stakerConfig.ts [hunk 1]
```diff
@@ -4,11 +4,23 @@ import {
...
 
+// Cache
+
+const STAKER_ITEM_METADATA = "staker-item-metadata";
+
+export const stakerItemMetadata = dbCache.indexedByKey<StakerItemData, string>({
+  rootKey: STAKER_ITEM_METADATA,
+  getKey: target => target,
+  validate: (id, metadata) =>
+    typeof id === "string" && typeof metadata === "object"
```
Comment (dsimog01) [inline packages/dappmanager/src/db/stakerConfig.ts:21]: Maybe you can specify more the type of metadata?

File: packages/dappmanager/src/db/stakerConfig.ts [hunk 2]
```diff
@@ -4,11 +4,23 @@ import {
...
-import { dbMain } from "./dbFactory";
+import { dbCache, dbMain } from "./dbFactory";
 import { interceptGlobalEnvOnSet } from "./interceptGlobalEnvOnSet";
 
+// Cache
+
+const STAKER_ITEM_METADATA = "staker-item-metadata";
+
+export const stakerItemMetadata = dbCache.indexedByKey<StakerItemData, string>({
+  rootKey: STAKER_ITEM_METADATA,
```
Comment (dsimog01) [inline packages/dappmanager/src/db/stakerConfig.ts:18]: Why don't you just put "staker-item-metadata" here? It seems you don't use the constant in any other place.

Comment (pablomendezroyo) [reply packages/dappmanager/src/db/stakerConfig.ts:18]: Checkout the rest of db files, they use that standard. So in order to not break the standard I keep it like that

File: packages/dappmanager/src/db/stakerConfig.ts (hunk 1 above)
Comment (pablomendezroyo) [reply packages/dappmanager/src/db/stakerConfig.ts:21]: the js function `typeof` is meant for native js types: `string` `boolean` `number`, etc.

File: packages/dappmanager/src/calls/packageRemove.ts [hunk 3]
```diff
@@ -87,7 +87,15 @@ export async function packageRemove({
     const containerNames = dnp.containers.map(c => c.containerName);
     await Promise.all(
       containerNames.map(async containerName => {
-        await dockerContainerStop(containerName, { timeout });
+        // Continue removing package even if container is already stopped
+        await dockerContainerStop(containerName, { timeout }).catch(e => {
+          if (
+            e.reason.includes("container already stopped") &&
```
[the remark this answers is not in the document]
Comment (pablomendezroyo) [reply packages/dappmanager/src/calls/packageRemove.ts:93]: Agree!

**Pull request merged**